nutils/util.py

```diff
@@ -512,6 +514,23 @@ def set_current(*args, **kwargs):
     return set_current
 
 
+@contextlib.contextmanager
+def timeit():
+    '''Context that logs begin time, end time, and duration.'''
+
+    t0 = time.time()
+    fmt = '%Y-%m-%d %H:%M:%S'
+    treelog.info(f'start {time.strftime(fmt,time.localtime(t0))}')
+    try:
+        yield
+    finally:
+        te = time.time()
+        m, s = divmod(int(te - t0), 60)
+        if m > 60:
+            m = '{}:{:02d}'.format(*divmod(m, 60))
+        treelog.info(f'finish {time.strftime(fmt,time.localtime(t0))}, elapsed {m}:{s:02d}')
```

Comment:
Alternatively you can use `datetime`. Advantage: you can write the time format of a `datetime` as format parameter:

```python
t0 = datetime.now()
treelog.info(f'start {t0:%Y-%m-%d %H:%M:%S}')
```

The difference between two `datetime` instances, `te - t0`, has a `total_seconds()` method.